pkg/controllers/routing/bgp_policies.go

```diff
@@ -45,9 +45,17 @@ func (nrc *NetworkRoutingController) AddPolicies() error {
 	// creates prefix set to represent all the advertisable IP associated with the services
 	advIPPrefixList := make([]config.Prefix, 0)
 	advIps, _, _ := nrc.getAllVIPs()
-	for _, ip := range advIps {
-		advIPPrefixList = append(advIPPrefixList, config.Prefix{IpPrefix: ip + "/32"})
+
+	//If the value of advertise-service-cluster-ip-range parameter is not empty, then the value of advertise-service-cluster-ip-range parameter is put into RIB, otherwise it will be done according to the original rules.
```

Comment:
I would like both service CIDR and service VIP to be announced **only** for the services marked with external traffic policy set to local. So that on upstream routers with give /32 router precedence.

Reply:
@cloudnativer Did you give any thought to above comment? I see below three scenarios

- if `--advertise-service-cluster-ip-range` is configured then advertise **ONLY** the service cluster IP range from the nodes and **DO NOT** advertise service VIP's from the node (which is what this PR intends to achieve) irrespective of fact service has pod's running on the node and service is marked `externalTrafficPolicy=Local`.
- if `--advertise-service-cluster-ip-range` is configured then advertise the service cluster IP range from the nodes **AND** advertise `/32` service VIP's from the node if node has a endpoint pod corresponding to the service running on the node and service is marked `externalTrafficPolicy=Local`.
- if `--advertise-service-cluster-ip-range` is **NOT** configured then keep the original behaviour i.e.) advertise `/32` VIP from all the nodes if service is not marked with `externalTrafficPolicy=Local` and advertise service VIP **ONLY** from the nodes which has a endpoint pod corresponding to the service is running on the node and service is marked with `externalTrafficPolicy=Local` 

problem with #1 is it can blackhole the traffic for services set to `externalTrafficPolicy=Local`. Traffic to service VIP get's ECMP ed to a node not running any service endpoint pod and proxy running on the node will reject the traffic.

Reply:
> @cloudnativer Did you give any thought to above comment? I see below three scenarios
> 
> * if `--advertise-service-cluster-ip-range` is configured then advertise **ONLY** the service cluster IP range from the nodes and **DO NOT** advertise service VIP's from the node (which is what this PR intends to achieve) irrespective of fact service has pod's running on the node and service is marked `externalTrafficPolicy=Local`.
> * if `--advertise-service-cluster-ip-range` is configured then advertise the service cluster IP range from the nodes **AND** advertise `/32` service VIP's from the node if node has a endpoint pod corresponding to the service running on the node and service is marked `externalTrafficPolicy=Local`.
> * if `--advertise-service-cluster-ip-range` is **NOT** configured then keep the original behaviour i.e.) advertise `/32` VIP from all the nodes if service is not marked with `externalTrafficPolicy=Local` and advertise service VIP **ONLY** from the nodes which has a endpoint pod corresponding to the service is running on the node and service is marked with `externalTrafficPolicy=Local`
> 
> problem with #1 is it can blackhole the traffic for services set to `externalTrafficPolicy=Local`. Traffic to service VIP get's ECMP ed to a node not running any service endpoint pod and proxy running on the node will reject the traffic.


This problem does exist if ·externalTrafficPolicy=Local· is set.

(1) The applicable conditions of `advertise-service-cluster-ip-range` are as follows:
 * `externalTrafficPolicy ` can only be used when the service is set to loadbalancer or nodeport.
 * If the service is set to loadbalancer or nodeport, you should set `externalTrafficPolicy=Cluster` at the same time, so that the `advertise-service-cluster-ip-range` parameter is meaningful.
 * Except for this situation, you can directly set up `advertise-service-cluster-ip-range`.

(2)In large-scale environment, network traffic load balancing is very important. However, the configuration of `externalTrafficPolicy=Local` will lead to unbalanced network traffic load, so our production environment uses the default `externalTrafficPolicy=Cluster`.